src/processor/processor_task.cpp

```diff
@@ -38,12 +38,26 @@ std::unique_ptr<ResultSet> ProcessorTask::populateResultSet(
             auto expression = dataChunkDescriptor->getExpression(j);
             auto vector =
                 std::make_shared<common::ValueVector>(expression->dataType, memoryManager);
+            if (vector->dataType.getTypeID() == common::STRUCT) {
+                addStructFieldsVectors(vector.get(), dataChunk.get(), memoryManager);
+            }
             dataChunk->insert(j, std::move(vector));
         }
         resultSet->insert(i, std::move(dataChunk));
     }
     return resultSet;
 }
 
+void ProcessorTask::addStructFieldsVectors(common::ValueVector* structVector,
```

Comment:
Add a TODO for me to move this logic into front-end

Reply:
done